geowebcache/core/src/main/java/org/geowebcache/seed/TruncateAllRequest.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Lesser General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU Lesser General Public License along with this
+ * program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Imran Rajjad / Geosolutions 2019
+ */
+package org.geowebcache.seed;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+import java.io.Serializable;
+import java.util.Iterator;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geowebcache.GeoWebCacheException;
+import org.geowebcache.layer.TileLayer;
+import org.geowebcache.storage.StorageBroker;
+import org.geowebcache.storage.StorageException;
+
+/** @author ImranR */
+@XStreamAlias("truncateAll")
+public class TruncateAllRequest implements MassTruncateRequest, Serializable {
+
+    /** serialVersionUID */
+    private static final long serialVersionUID = -4730372010898498464L;
+
+    private static final Log log = LogFactory.getLog(TruncateAllRequest.class);
+
+    @XStreamOmitField private StringBuilder trucatedLayers = new StringBuilder();
+
+    @Override
+    public boolean doTruncate(StorageBroker sb, TileBreeder breeder)
+            throws StorageException, GeoWebCacheException {
+        Iterator<TileLayer> iterator = breeder.getLayers().iterator();
+        TileLayer toTruncate;
+        boolean truncated = false;
+        while (iterator.hasNext()) {
+            toTruncate = iterator.next();
+            truncated = sb.delete(toTruncate.getName());

```

Comment:
One more thing, does disk quota on the layer in question work fine after this call? I'm asking because GeoServer does the layer truncation by iterating on its gridset ids, but I see the TruncateLayer class here also does it this way.
Wondering if listeners might be thrown off thinking the layer is gone, rather than its tiles, have a quick look please.

Reply:
@aaime 
I truncated layers and again seeded them back and forth from UI and REST. After clearing all layers and then asking GWC to seed the layer again went fine both with standalone and integrated GWC. 

So to test your concerns I will follow the below steps:

- Seed some layers
- Nuke the cache
- Seed the layers again and see if there are any issues

Would that do?

Reply:
Do the same with disk quota enabled.

Reply:
@aaime 

testing with diskquota shows these logs

` (GWC DiskQuota Store Writer-1) INFO [bdb.BDBQuotaStore] - Can't add usage stats. TileSet does not exist. Was it deleted? FX_LP_VT#EPSG:4326#image/png8`

is this related to your concerns?

Reply:
Yes, that is a serious problem. Please rewrite the delete code mimicking the geoserver one, deleting by gridset. If you can, please fix TruncateLayer as well.
See: https://github.com/geoserver/geoserver/blob/ac0ed118b8a316de7cd0d8bee6d4d4919ac1e537/src/gwc/src/main/java/org/geoserver/gwc/GWC.java#L356
Then check disk quota works fine, reports zero usage on all layers, and starts collectin stats again as new tiles are getting created.

Reply:
will do